src/main.rs

```diff
@@ -0,0 +1,296 @@
+        );
+    }
+
+    Ok(result)
+}
+
+/// Checks options in the same target for duplicate keys
+fn ensure_no_duplicate_keys(grouped: &NamespaceMap) -> Result<()> {
+    for (_, targets) in grouped {
+        for (_, filedata) in targets {
+            let mut key_to_file = HashMap::<String, String>::new();
+            for FileData { path, data } in filedata {
+                for key in data.keys() {
+                    if let Some(first_file) = key_to_file.get(key) {
+                        return Err(AppError::DuplicateKey {
+                            key: key.to_string(),
+                            first_file: first_file.to_string(),
+                            second_file: path.to_string(),
+                        });
+                    }
+                    key_to_file.insert(key.to_string(), path.to_string());
+                }
+            }
+        }
+    }
+    Ok(())
+}
+
+/// Merges keys from many option files into one map
+fn merge_keys(filedata: &[FileData]) -> OptionsMap {
+    let mut result = HashMap::new();
+    for FileData { data, .. } in filedata {
+        for (key, value) in data {
+            result.insert(key.clone(), value.clone());
+        }
+    }
+    result
+}
+
+/// Generates the list of output JSON files
+/// Uses the default target and handles overrides from other targets
+fn generate_json(maps: NamespaceMap) -> Result<Vec<(String, String)>> {
+    let mut json_outputs: Vec<(String, String)> = Vec::new();
+
+    // merge files in the same target together
+    for (namespace, targets) in maps {
+        let default_target = targets.get("default").ok_or_else(|| {
+            AppError::Validation(format!(
+                "Namespace '{}' is missing required 'default' target",
+                namespace
+            ))
+        })?;
+        let defaults = merge_keys(default_target);
+
+        for (target, filedatas) in targets {
+            let mut merged = defaults.clone();
+            merged.extend(merge_keys(&filedatas));
+
+            // Convert to BTreeMap for sorted keys
+            let sorted_merged: BTreeMap<_, _> = merged.into_iter().collect();
+
+            let mut with_option_key = BTreeMap::new();
+            with_option_key.insert("options", sorted_merged);
+            json_outputs.push((
+                format!("sentry-options-{namespace}-{target}.json"),
+                serde_json::to_string_pretty(&with_option_key)?,
+            ));
+        }
+    }
+    Ok(json_outputs)
+}
+
+/// Writes JSON data to JSON files in the specified directory
+fn write_json(out_path: PathBuf, json_outputs: Vec<(String, String)>) -> Result<()> {
+    fs::create_dir_all(&out_path)?;
+
+    for (filename, json_text) in json_outputs {
+        let filepath = out_path.join(&filename);
+        fs::write(&filepath, json_text)?;
+    }
+    Ok(())
+}
+
+fn main() -> Result<()> {
+    let args = Args::parse();
+
+    let out_path = PathBuf::from(&args.out);
+    if out_path.exists() {
```

Comment:
do we even need to check for existence of output directory? do we want to just overwrite files?

Reply:
when would the output directory already exist?